src/coreclr/vm/ilmarshalers.cpp

```diff
@@ -2261,7 +2295,22 @@ void ILLayoutClassPtrMarshalerBase::EmitClearNativeTemp(ILCodeStream* pslILEmit)
     }
 }
 
+bool ILLayoutClassPtrMarshalerBase::EmitExactTypeCheck(ILCodeStream* pslILEmit, ILCodeLabel* isNotMatchingTypeLabel)
```

Comment:
Curious - would it be possible to make this a static local function to this compilation unit and not expose it on the type? If it is even marginally complicated leave as-is.

Reply:
We'd have to add an extra parameter, which wouldn't be too bad.

I added it as a member of the type since it's only used in `ILLayoutClassPtrMarshalerBase` and derived classes, so I felt that putting it on the class as a protected member already scoped it pretty well.

Reply:
That makes sense.